improver_tests/snow_fraction/test_SnowFraction.py

```diff
@@ -0,0 +1,161 @@
+    )
+    snow = set_up_variable_cube(
+        snow_data,
+        name.format(phase="lwe_snow"),
+        units,
+        time_bounds=time_bounds,
+        spatial_grid="equalarea",
+        attributes=COMMON_ATTRS,
+    )
+    return rain, snow
+
+
+@pytest.mark.parametrize(
+    "cube_name", ("{phase}rate", "thickness_of_{phase}fall_amount")
+)
+def test_basic(cube_name):
+    """Run a test with four values, including one that will trigger divide-by-zero.
+    Check data and metadata of result."""
+    rain, snow = setup_cubes(name=cube_name)
+
+    expected_data = np.array([[1.0, 1.0], [2.0 / 3.0, 0.0]], dtype=np.float32)
+    result = SnowFraction()(iris.cube.CubeList([rain, snow]))
+    assert isinstance(result, iris.cube.Cube)
+    assert str(result.units) == "1"
+    assert result.name() == "snow_fraction"
+    assert result.attributes == COMMON_ATTRS
+    assert np.allclose(result.data, expected_data)
+
+
+def test_acclen_mismatch_error():
+    """Test the process function with mismatched accumulation lengths"""
+    rain, snow = setup_cubes(name="thickness_of_{phase}fall_amount")
+    time_coords = construct_scalar_time_coords(
+        [c.point for c in snow.coord("time").cells()],
+        (datetime(2017, 11, 10, 1, 0), datetime(2017, 11, 10, 4, 0)),
+        snow.coord("forecast_reference_time").cell(0).point,
+    )
+    _ = [snow.replace_coord(coord) for coord, _ in time_coords]
+    with pytest.raises(
+        ValueError, match="Rain and Snow cubes do not have the same time coord"
+    ):
+        SnowFraction()(iris.cube.CubeList([rain, snow]))
+
+
+def test_dims_mismatch_error():
+    """Test the process function with mismatched dimensions"""
+    rain, snow = setup_cubes(snow_data=np.array([[0, 0.5]], dtype=np.float32))
+    with pytest.raises(
+        ValueError, match="Rain and Snow cubes are not on the same grid"
+    ):
+        SnowFraction()(iris.cube.CubeList([rain, snow]))
+
+
+def test_missing_cube_error():
+    """Test the process function with one cube missing"""
+    rain, _ = setup_cubes()
+    with pytest.raises(ValueError, match="Expected exactly 2 input cubes, found 1"):
+        SnowFraction()(iris.cube.CubeList([rain]))
+
+
+def test_wrong_input_names_error():
+    """Test the process function with incorrect input cubes"""
+    rain, snow = setup_cubes()
+    rain.rename("kittens")
+    snow.rename("puppies")
+    with pytest.raises(
+        ValueError,
+        match="Could not find both rain and snow in \['kittens', 'puppies'\]",
+    ):
+        SnowFraction()(iris.cube.CubeList([rain, snow]))
+
+
+def test_input_name_matches_both_phases_error():
+    """Test the process function with an input cube that has both rain and snow in its name"""
+    rain, snow = setup_cubes()
+    rain.rename("its_raining_snowy_kittens")
+    snow.rename("puppies")
+    with pytest.raises(
+        ValueError,
+        match="Failed to find unique Rain and Snow cubes from \['its_raining_snowy_kittens', 'puppies'\]",
```

Comment:
This is causing your unit test to fail at the moment as you've just un(de)-capitalised in the plugin.

Reply:
[rude-noise]. I've done this properly now. All instances of "Rain and Snow" are now capitalised in the way I was taught at primary school.